modules/system-view/src/test/java/org/apache/ignite/internal/systemview/SystemViewManagerTest.java

```diff
@@ -0,0 +1,130 @@
+import static org.apache.ignite.internal.testframework.IgniteTestUtils.assertThrowsWithCause;
+import static org.apache.ignite.internal.testframework.matchers.CompletableFutureExceptionMatcher.willThrowFast;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.aMapWithSize;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.Mockito.only;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import org.apache.ignite.internal.catalog.CatalogManager;
+import org.apache.ignite.internal.testframework.BaseIgniteAbstractTest;
+import org.apache.ignite.internal.util.AsyncCursor;
+import org.apache.ignite.lang.NodeStoppingException;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+/**
+ * Test class for {@link SystemViewManagerImpl}.
+ */
+@ExtendWith(MockitoExtension.class)
+public class SystemViewManagerTest extends BaseIgniteAbstractTest {
+    @Mock
+    private CatalogManager catalog;
+
+    @Test
+    public void registerDuplicateNameFails() {
+        SystemViewManager viewMgr = new SystemViewManagerImpl(catalog);
+
+        String name = "testView1";
+
+        ClusterSystemView<Object> view = dummyView(name);
+        ClusterSystemView<Object> viewWithSameName = dummyView(name);
+
+        viewMgr.register(view);
+
+        assertThrows(IllegalArgumentException.class, () -> viewMgr.register(viewWithSameName));
+        verifyNoInteractions(catalog);
+    }
+
+    @Test
+    public void registerAfterStartFails() {
+        SystemViewManager viewMgr = new SystemViewManagerImpl(catalog);
+
+        viewMgr.start();
+
+        assertThrows(IllegalStateException.class, () -> viewMgr.register(dummyView("test")));
+        verifyNoInteractions(catalog);
+    }
+
+    @Test
+    public void nodeAttributesUpdatedAfterStart() {
+        Mockito.when(catalog.execute(anyList())).thenReturn(completedFuture(null));
+
+        String name1 = "view1";
+        String name2 = "view2";
+        SystemViewManagerImpl viewMgr = new SystemViewManagerImpl(catalog);
+
+        viewMgr.register(dummyView(name1));
+        viewMgr.register(dummyView(name2));
+
+        assertThat(viewMgr.nodeAttributes(), aMapWithSize(0));
+
+        viewMgr.start();
+
+        verify(catalog, only()).execute(anyList());
+        verifyNoMoreInteractions(catalog);
+
+        assertThat(viewMgr.nodeAttributes(), is(Map.of(NODE_ATTRIBUTES_KEY, String.join(NODE_ATTRIBUTES_LIST_SEPARATOR, name1, name2))));
+    }
+
+    @Test
+    @SuppressWarnings("ThrowableNotThrown")
```

Comment:
I think `@SuppressWarnings("ThrowableNotThrown")` can be drop because Component::stop throws Exception?